### DISCUSSION TITLE
Allow configurable concurrency setting.

### DISCUSSION THREAD

**Discussion by andychongyz** [graphiti-api/graphiti#418] [Ideas]
Hi people, I realize the gem automatically turns on the concurrency feature by default on production. And we have been suffered from the ActiveRecord Connection Pool issue until we monkey-patched the `establish_concurrency` method. What if we allow setting that in the `configure` block. I also realized that concurrency does not benefits smaller app running on a single core server like the Heroku Standard-1X instance.

Here's a small snapshot of before and after (the small rocket icon indicates the deployment to turn off concurrency):
<img width="1656" alt="image" src="https://user-images.githubusercontent.com/12697240/170174247-904ad8c8-7074-4e5d-a89f-ad5dc63f3c8c.png">

**Comment by richmolj** [comment]
Yup it's already there - `config.concurrency = false`

**Reply by andychongyz** [reply]
But, that doesn't work for me. I have the configuration file in the /initializers dir in my rails app. But the `establish_concurrency` method executes after the configuration code is executed, which overwrite the configuration. Could it be because I'm on Ruby 3 🤔 .
I ended up monkey-patched the code like below:
```ruby
::Graphiti::Rails::Railtie.class_eval do
  def establish_concurrency
    Graphiti.config.concurrency = false
  end
end
```
